content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
+The security objective is to contain spam activity in a GossipSub network by enforcing a global messaging rate to all the peers.
+Peers that violate the messaging rate are considered spammers and their message is considered spam.
+Spammers are also financially punished and removed from the system. 
 
 
 **Protocol identifier***: `/vac/waku/waku-rln-relay/2.0.0-alpha1`
 
 # Motivation
 
-In open p2p messaging networks, one big problem is spam-resistance. 
-Existing solutions, such as Whisper’s proof of work, are insufficient, especially for heterogeneous nodes. 
+In open and anonymous p2p messaging networks, one big problem is spam resistance. 
+Existing solutions, such as Whisper’s proof of work are computationally expensive hence not suitable for resource-limited nodes. 
 Other reputation-based approaches might not be desirable, due to issues around arbitrary exclusion and privacy.
 
-We augment the `relay` protocol with a novel, light, and effective spam prevention mechanism which also suits the resource-constrained nodes.
-
-<!-- TODO: Fill in more -->
+We augment the [`11/WAKU2-RELAY`](/spec/11) protocol with a novel construct of [RLN](/spec/32) to enable an efficient economic spam prevention mechanism that can be run in resource-constrained environments.
 
 
 # Flow
+The messaging rate is set to 1 per `epoch`, 
+where `epoch` indicates the number of `T` seconds elapsed since the Unix epoch event.
+The value of `T` is application-dependent.
+See section [Recommended System Parameters](#recommended-system-parameters) for some recommended ways to set a sensible `T` value.
+Peers subscribed to a spam-protected `pubsubTopic` are only allowed to send one message per `epoch`
+(messaging rate MAY be enforced for `WakuMessages` with a specific `contentTopic` published on a `pubsubTopic`, but that is an implementation choice).
+
 ## SetUp and Registration
-A peer willing to publish a message is required to register. 
+Peers subscribed to a specific `pubsubTopic` form a [rln group](/spec/32).
+<!-- link to the rln group definition in the rln RFC -->
+Peers MUST be registered to the rln group to be able to publish messages.
 Registration is moderated through a smart contract deployed on the Ethereum blockchain. 
-The state of the contract contains the list of registered members. 
+Each peer has an [rln key pair](/spec/32) denoted by `sk`  and `pk`.
+The `sk` is secret data and MUST be persisted securely by the peer.
+The state of the membership contract contains the list of registered members' public identity keys i.e., `pk`s. 
+For the registration, a peer creates a transaction that invokes the registration function of the contract via which registers its `pk` in the group. 
+The transaction also transfers  `v` amount of ether to the contract to be staked. 
+`v` is a system parameter.
```

Comment:
Could we make this list of system parameters more clear? E.g. with a list, a table or a paragraph for each

Reply:
sure, added a table under the "Recommended System Parameters".